app/src/main/res/layout/fragment_main.xml

```diff
@@ -1,103 +1,103 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```

Comment:
Overall, the layout seems okay, works as expected, but there are few things that can be improved:
1. You've added horizontal guideline, which is good. However, you put wrong constraints for the ImageViews above (`qq_flag` and `ru_flag`). You should align the bottoms of the TextViews (`qq_text` and `ru_text`) to the top of guideline. Next, you align the ImageViews bottom to the top of TextView.
2. You don't need any margins around ImageViews, instead you can chain them horizontally, that way it will automatically allocate equal space around views, look at [official documentation](https://developer.android.com/reference/android/support/constraint/ConstraintLayout#Chains) and this article: https://constraintlayout.com/basics/create_chains.html
3. Do not use left/right in layout parameters like `app:layout_constraintLeft_toRightOf`, instead you need use start/end, that way the above will become app:layout_constraintStart_toEndOf. Using right/left in layouts are outdated these days, the main reason is some of the languages like arabic/hebrew are written from right to left, for those languages our left/right will be opposite, that's why google introduced start/end, which will be based on language specifics.

Reply:
Why we need chains? I have removed margins but it also works properly without margins and chains.

Reply:
Okay, looks good. Probably we don't need chains here as we have static content here.